configuration/secrets.rst

```diff
@@ -0,0 +1,378 @@
+This can be done with the provided command ``secrets:generate-keys``.
+
+.. code-block:: terminal
+
+    $ APP_ENV=prod php bin/console secrets:generate-keys
+
+This command generates a new pair of asymetric **encryption keys** in
+``%kernel.project_dir%/config/secrets/%kernel.environment%``.
+The **encryption key** stored in ``%kernel.environment%.sodium.encrypt.key``
+file is used to encrypt secrets when developer adds or updates secrets. This
+key can be commited and publicly shared. The **decryption key** stored in
+``%kernel.environment%.sodium.decrypt.key`` file will be used to decrypt secrets
+and inject the revealed values in services. The number of people who possess
+this key should be as small as possible.
+
+.. caution::
+
+    The ``prod.sodium.decrypt.key`` file is sensitive and **should not** be
+    committed nor publicly shared. Your team developers and Continuous
+    Integration services don't need that key. If the **decryption key** have
+    been exposed (ex-employee leaving for instance), you should consider to
+    generate a new one with the command ``secrets:generate-keys --rotate``.
+
+.. _secrets-set:
+
+Create or Update Secrets
+------------------------
+
+You can add new secrets with the command ``secrets:set``. Symfony will ask you,
+in a hidden prompt, to enter the value. Symfony will encrypt this value and
+store it in a file located by default in the folder
+``%kernel.project_dir%/config/secrets/%kernel.environment%``.
+This file should be committed along side the other project's files.
+
+.. code-block:: terminal
+
+    # create a "DATABASE_PASSWORD" secret interactively
+    $ php bin/console secrets:set DATABASE_PASSWORD
+
+    # create secret for the "prod" environment
+    $ APP_ENV=prod php bin/console secrets:set DATABASE_PASSWORD
+
+    # provide a file where to read the secret from
+    $ php bin/console secrets:set APPLICATION_CREDENTIAL ~/Download/key.json
+
+    # or contents passed to STDIN
+    $ echo -n "$AWS_SECRET_ACCESS_KEY" | php bin/console secrets:set AWS_KEY -
+
+    # or let Symfony generate a random one for you
+    $ php bin/console secrets:set REMEMBER_ME --random
+
+If the secret already exists, it value will be overridden by the new one.
+
+.. tip::
+
+    the ``--random```flag will display the generated password in the response.
+    you can use linux pipe and ``xclip`` to store it directly in your clipboard
+    in order to avoid to display it
+
+    .. code-block:: terminal
+
+        $ php bin/console secrets:set REMEMBER_ME --random | xclip -selection c
+
+.. _secrets-reference:
+
+Referencing Secrets in Configuration Files
+------------------------------------------
+
+You can reference the secrets in any configuration option enclosing their names
+using the **secret** :ref:`environment variable processors <env-var-processors>`.
+Their actual values will be resolved at runtime, so that container compilation
+and cache warmup don't need the **decryption key**.
```

Comment:
Oh: we should provide a way to dump a vault as a`.env.local` file at build time, then `composer dump-env`.
That would prevent putting the decryption key on running server + bypass decrypt altogether!